util/include/thread_pool.hpp

```diff
@@ -73,11 +74,24 @@ class ThreadPool {
     {
       auto lock = std::lock_guard{task_queue_.mutex};
       task_queue_.tasks.push(std::move(task));
+      auto done_lock = std::lock_guard{task_queue_.done_check_mutex};
```

Comment:
Yes I will change this and make it consistent.

Reply:
I reviewed the threadpool again, its actually consistent. Also auto lock = locktype{mutex} is a syntactic sugar for locktype lock(mutex) since both are initialisation. I will discuss and then change this one.